src/ApplePayGateway.php

```diff
@@ -0,0 +1,206 @@
+ *    ))->send();
+ *
+ *    // getApplePayPaymentSessionObject() can then be called to retrieve the ApplePayPaymentSession object.
+ *    // You can ensure the response is successful before sending it to the front end. If there is an error the
+ *    // response body will be empty.
+ *    if ($authorizeResponse->isSuccessful()) {
+ *        echo 'Status Code: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *        echo 'Status Reason: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *        echo 'Apple Pay Payment Session Object: ' . $authorizeResponse->getPaymentSessionObject()
+ *          . PHP_EOL;
+ *        echo 'Response: ' . $authorizeResponse->getResponse() . PHP_EOL;
+ *    } else {
+ *        // Error handling.
+ *        echo 'ERROR: Request not successful.';
+ *        echo 'Status Code: ' . $authorizeResponse->getCode() . PHP_EOL;
+ *        echo 'Status Message: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *    }
+ *
+ *    //Pass authorizeResponse back to the client to validate your merchant and continue with an Apple Pay payment.
+ *    //If successful, the payment sheet should be fully loaded.
+ *    $apple_pay_session = $authorizeResponse->getPaymentSessionObject();
+ *
+ *     // TODO: Add functionality for completeAuthorize() and capture().
+ *     // An opaque Apple Pay Session is returned as a response (expires after 5 mins) and it can be sent to
+ *     // the front end to fully load the payment sheet. This allows the user to optionally configure their
+ *     // payment option and shipping methods (if needed) and submit their payment.
+ * </code>
+ */
+class ApplePayGateway extends AbstractVindiciaGateway
+{
+
+    /**
+     * Get the gateway name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return 'Vindicia ApplePay';
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getPemCertPath()
+    {
+        return $this->getParameter('pemCertPath');
+    }
+
+    /**
+     * A certificate associated with your merchant ID.
+     * Example:
+     * setPemCertPath('./certs/path_to_cert.crt.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setPemCertPath($value)
+    {
+        return $this->setParameter('pemCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPath()
+    {
+        return $this->getParameter('keyCertPath');
+    }
+
+    /**
+     * A key certificate associated with your merchant ID.
+     * Example:
+     * setKeyCertPath('./certs/path_to_cert.key.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setKeyCertPath($value)
+    {
+        return $this->setParameter('keyCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPassword()
+    {
+        return $this->getParameter('keyCertPassword');
```

Comment:
Could you make all of these strings into consts (keyCertPassword, keyCertPassword, pemCertPath)

Reply:
I don't think that would be necessary since there's a lot of them, mostly only used twice. Also, since this is an extension of the Omnipay-Common library, I believe we're trying to keep the same style format as the original library as well as the other classes in this repo.

Omnipay-Common found here: https://github.com/thephpleague/omnipay-common